ripgrep/releases.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+// this may need customizations between packages
+const osMap = {
+  macos: /\b(mac|darwin|iPhone|iOS|iPad)/i,
+  linux: /\b(linux)/i,
+  win: /\b(win|microsoft|msft)/i,
+  sunos: /\b(sun)/i,
+  aix: /\b(aix)/i
+};
+
+const archMap = {
+  amd64: /(amd64|x64|[_\-]64)/i,
+  x86: /\b(x86)(?![_\-]64)/i,
+  ppc64le: /\b(ppc64le)/i,
+  ppc64: /\b(ppc64)\b/i,
+  i686: /\b(i686)\b/i,
+  arm64: /\b(arm64|arm)/i,
+  armv7l: /\b(armv?7l)/i,
+  armv6l: /\b(armv?6l)/i,
+  s390x: /\b(s390x)/i
+};
+
+const fileExtMap = {
+  deb: /\.deb$/i,
+  pkg: /\.pkg$/i,
+  exe: /\.exe$/i,
+  msi: /\.msi$/i,
+  zip: /\.zip$/i,
+  tar: /\.(tar(\.?(gz)?)|tgz)/i,
```

Comment:
This is too strict for the general case.
There's bz2, gz, xz, and probably others.